### PR TITLE
(DOCSP-25707): realm-kotlin@1.2.0: Change user.identity to user.id

### PR DESCRIPTION
**Pull request by mongomoe**
### Jira

- https://jira.mongodb.org/browse/DOCSP-25707

### Staged Changes (Requires MongoDB Corp SSO)

- [QuickStart - Kotlin](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/realmresults-query-kotlin/sdk/kotlin/quick-start/)
- [Register Users - Kotlin](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/realmresults-query-kotlin/sdk/kotlin/app-services/register-users/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/QuickStartTest.kt [hunk 1]
```diff
@@ -101,11 +101,11 @@ class QuickStartTest: RealmTest() {
             //     user,
             //     setOf(Item::class)
             // ) // the SyncConfiguration defaults to Flexible Sync, if a Partition is not specified
-            //     .initialSubscriptions { realm ->
+            //     .initialSubscripti.ons { realm ->
```
Comment (mongodben) [inline examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/QuickStartTest.kt:104]: this seems like a typo

File: source/examples/generated/kotlin/QuickStartTest.snippet.quick-start.kt [hunk 2]
```diff
@@ -11,11 +11,11 @@ runBlocking {
         user,
         setOf(Item::class)
     ) // the SyncConfiguration defaults to Flexible Sync, if a Partition is not specified
-        .initialSubscriptions { realm ->
+        .initialSubscripti.ons { realm ->
```
Comment (mongodben) [inline source/examples/generated/kotlin/QuickStartTest.snippet.quick-start.kt:14]: ditto above abt typo

**Review by mongodben — COMMENTED** [review]
this seems like a breaking change, should there be a note about having ot change all "identity" to "id" somewhere?

**Comment by mongomoe** [comment]
> this seems like a breaking change, should there be a note about having ot change all "identity" to "id" somewhere?

It's not actually a breaking change, ``user.identity`` is still technically usable, it's just deprecated. (See: [Realm-Kotlin 1.2.0 Changelog](https://github.com/realm/realm-kotlin/blob/master/CHANGELOG.md#120-2022-09-30)

**Review by mongodben — APPROVED** [review]
lgtm!

**Pull request merged**